### ISSUE TITLE
BUG: convergence failure with llnull in Poisson

### ISSUE THREAD

**Issue by josef-pkt** [statsmodels/statsmodels#3476]
see mailing list Poisson example
We can work around convergence failure with default settings in the main model, but we cannot influence the fit of the null model if that doesn't converge.

https://groups.google.com/d/msg/pystatsmodels/FAnjBe_RynE/e8hWinbZCQAJ

We can compute the constant directly from the data without optimization in cases like Poisson, and I guess most other GLM type models, maybe not when offset and exposure are used. And I'm not sure about noncanonical links (which might need a detour)

**Comment by josef-pkt** [comment]
Note we can have an explicit solution for the parameter constant only model.

something like this

```
const = np.exp(params) = endog.mean() / np.exp(offset + exposure)
params = [np.log(const]
```

AFAIR, we still don't have a function to compute the negative binomial dispersion function/parameters from the residuals.

**Comment by josef-pkt** [comment]
fixed in #3921 I used the mailing list example as unit test case
